### PR TITLE
Work around webpack trying to bundle react-native

### PR DESCRIPTION
**Pull request by b123400**
connect https://github.com/SkygearIO/skygear-SDK-JS/issues/160

This PR is not exactly doing what the issue is saying, but should solve the problem.

### CODE REVIEW

**Comment by oursky-travis** [comment]
@b123400, thanks for your PR! By analyzing the history of the files in this pull request, we identified @limouren to be a potential reviewer.

**Comment by louischan-oursky** [comment]
I tried this patch but it seems that it does not work with react native packager.

Here are the steps

1. Build this patch and [commit the build](https://github.com/louischan-oursky/skygear-SDK-JS/commits/require-react-native)
2. Initialize a react native project and configure `skygear` to point to the above commit
3. Run the react native app

It should fail with the following log

```
2017-04-26 19:09:26.808690 MyApp[16449:6446190] 'Failed to get user', { [ReferenceError: Can't find variable: document]
  line: 98986,
  column: 32,
  sourceURL: 'http://192.168.1.127:8081/index.ios.bundle?platform=ios&dev=false&minify=false' }
```

I went through the line and column in the log and discovered that the storage is incorrectly recognized as `CookieStorage` (used in lib/store.js) which makes use of `document` which does not exist in react native.

**Comment by b123400** [comment]
TLDR: I give up working around webpack.

-------

- webpack translates all modules' name into number, then require them by the numbers in run time. Something like this:

      require('a') // becomes require(1);
      require = id => {1:module_a()}[id]()

- React native is using webpack with aliasing plugin, it changes the name of a bundle according to `@providesModule`
  - `react-native` is aliased to `react-native-implementation`
- `require('react-native')` doesn't work because it is aliased
- By working around webpack, the "string to number" translation is missing, and it is still requiring `react-native` after built, so it does not work.
- Tried to `require('react-native-implementation')`, it works. But then got a message saying loading module with string does not work in production environment.
- I give up working around webpack
